src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java

```diff
@@ -150,17 +80,12 @@ private static void populateFromZip(final File map, final Map<String, URI> avail
         if (entry.getName().contains("games/") && entry.getName().toLowerCase().endsWith(".xml")) {
           final URL url = loader.getResource(entry.getName());
           if (url != null) {
-            try {
-              final boolean added = addToAvailableGames(
-                  new URI(url.toString().replace(" ", "%20")),
-                  availableGames,
-                  mapNamePropertyList);
-              if (added && map.getName().length() > 4) {
-                availableMapFolderOrZipNames
-                    .add(map.getName().substring(0, map.getName().length() - ZIP_EXTENSION.length()));
-              }
-            } catch (final URISyntaxException e) {
-              // only happens when URI couldn't be build and therefore no entry was added. That's fine
+            final boolean added = addToAvailableGames(
+                URI.create(url.toString().replace(" ", "%20")),
```

Comment:
I don't think URI.create properly mirrors the previous behaviour.
According to the docs it simply wrapps the URISyntaxException in an IllegalStateException which is unchecked

Reply:
Now that I'm reviewing it again: Why don't just use `URL#toURI`?

Reply:
Valid question, i think you're right. Let's do that as a follow-up? I think I am mostly looking for a review of the changes as presented, not necessarily how much further we can god : )    There are seriously a lot more cleanups that could happen. THe intended change here is to remove teh try/catch. For the current scope I'd like to avoid updating any logic to try and keep the risk a bit lower. It's just too easy to break things..